Classes/common/HTTP/CDTSessionCookieInterceptor.m

```diff
@@ -0,0 +1,147 @@
+//
+//  CDTSessionCookieInterceptor.m
+//
+//
+//  Created by Rhys Short on 08/09/2015.
+//  Copyright (c) 2015 IBM Corp.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import "CDTSessionCookieInterceptor.h"
+#import "CDTLogging.h"
+
+@interface CDTSessionCookieInterceptor ()
+
+@property (nonnull, strong, nonatomic) NSString *cookieRequestBody;
+@property (nonatomic) BOOL shouldMakeCookieRequest;
+@property (nullable, strong, nonatomic) NSString *cookie;
+@property (nonnull, nonatomic, strong) NSURLSession *urlSession;
+
+@end
+
+@implementation CDTSessionCookieInterceptor
+
+- (instancetype)initWithUsername:(NSString *)username password:(NSString *)password
+{
+    self = [super init];
+    if (self) {
+        _cookieRequestBody = [NSString stringWithFormat:@"name=%@&password=%@", username, password];
+        _shouldMakeCookieRequest = YES;
+        NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
```

Comment:
I'd make this an ephemeral session, given there's no need for caching or anything much really.